recipes/cloudwatch_agent_install.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: cloudwatch_agent_install
+#
+# Copyright 2013-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Download the cloudwatch agent's public key
+remote_file node['cfncluster']['cloudwatch']['public_key_path'] do
+  source node['cfncluster']['cloudwatch']['public_key_url']
+  not_if { ::File.exist?(node['cfncluster']['cloudwatch']['public_key_path']) }
+end
+
+# Set URLs used to download the package and expected signature based on platform
+package_url_prefix = "https://s3.#{node['cfncluster']['cfn_region']}.amazonaws.com/amazoncloudwatch-agent-#{node['cfncluster']['cfn_region']}"
```

Comment:
Do these urls exist in every region, including `cn-north-1` and `us-gov-west-1`?

Reply:
My assumption was that they did. I'll need to confirm that.

Reply:
verified via this script
```
#!/bin/bash -e -x
package_name=amazon-cloudwatch-agent

# Try to get cloudwatch agent's public key
public_key_url=https://s3.amazonaws.com/amazoncloudwatch-agent/assets/${package_name}.gpg
curl -I $public_key_url | grep '200 OK'

for region in `aws ec2 describe-regions | jq -r '.Regions[].RegionName'`  #  us-gov-west-1 us-gov-east-1 cn-north-1 cn-northwest-1
do
  # Try to get package and package signature
  for os in amazon_linux centos ubuntu
  do
    package_url="https://s3.${region}.amazonaws.com/amazoncloudwatch-agent-${region}/${os}/amd64/latest/${package_name}"
    if [[ $os = ubuntu ]]
    then
      package_suffix=deb
    else
      package_suffix=rpm
    fi
    package_url=${package_url}.${package_suffix}
    sig_url=${package_url}.sig
    for url in $package_url $sig_url
    do
      curl -I $url | grep '200 OK'
    done
  done
done
```

Reply:
For China the domain must be changed to s3.<region>.amazonaws.com.cn

Reply:
The following urls worked form EC2 instances running in cn-north-1 and cn-northwest-1:
```https://s3.amazonaws.com/amazoncloudwatch-agent/assets/amazon-cloudwatch-agent.gpg
https://s3.cn-north-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-north-1/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm
https://s3.cn-north-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-north-1/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm.sig
https://s3.cn-north-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-north-1/centos/amd64/latest/amazon-cloudwatch-agent.rpm
https://s3.cn-north-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-north-1/centos/amd64/latest/amazon-cloudwatch-agent.rpm.sig
https://s3.cn-north-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-north-1/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb
https://s3.cn-north-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-north-1/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb.sig
https://s3.cn-northwest-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-northwest-1/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm
https://s3.cn-northwest-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-northwest-1/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm.sig
https://s3.cn-northwest-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-northwest-1/centos/amd64/latest/amazon-cloudwatch-agent.rpm
https://s3.cn-northwest-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-northwest-1/centos/amd64/latest/amazon-cloudwatch-agent.rpm.sig
https://s3.cn-northwest-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-northwest-1/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb
https://s3.cn-northwest-1.amazonaws.com.cn/amazoncloudwatch-agent-cn-northwest-1/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb.sig```